src/modules/Basket/PaymentInfo.tsx

```diff
@@ -0,0 +1,175 @@
+}
+
+const PaymentInfo = ({
+  setSelectedBasketType,
+  orderData,
+  setOrderNumber,
+}: Props) => {
+  const { productsCount } = useBasketContext()
+  const [pay, setPay] = useState('cash')
+  const [voucher, setVoucher] = useState<Voucher>({ voucherKey: '' })
+  const { handleVoucher, isValid } = useVoucherMutation(voucher)
+  const [caution, setCaution] = useState(false)
+  // const [check, setCheck] = useState(false)
+  const { voucherDataFromStorage: isVoucherValid } = useVoucherFromStorage(
+    'isVoucherValid',
+    false,
+  )
+  const { handleSubmitOrder } = usePostOrderMutation({
+    orderData,
+    setOrderNumber,
+    setSelectedBasketType,
+  })
+
+
+
+  const { voucherDataFromStorage: voucherCodeFromStorage } =
+    useVoucherFromStorage('voucherCode', '')
+
+  const handleSubmitVoucher = () => {
+    
+    handleVoucher();
+  if(!isVoucherValid ){setCaution(true)}else{setCaution(false)}
+  }
+
+  const handleCloseBasket = () => {
+    setSelectedBasketType('basket')
+  }
+
+
+  useEffect(() => {
+    if (pay === 'online') {
+      orderData.paymentType = PaymentType.online;
+  } else if (pay === 'cash') {
+      orderData.paymentType = PaymentType.cash;
+  } else {
+      orderData.paymentType = PaymentType.terminal;
+}
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pay]);
+
+
+useEffect(() => {
+  orderData.code = voucher.voucherKey
+// eslint-disable-next-line react-hooks/exhaustive-deps
+}, [voucher]);
+
+
+useEffect(() => {
+  if(isVoucherValid){setCaution(false)} 
+
+}, [isVoucherValid]);
+
+  return (
+    <>
+    {console.log('caution ', caution)}
+      <DrawerHeader
+        display="flex"
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        <Text
+          cursor="pointer"
+          onClick={() => setSelectedBasketType('delivery')}
+          fontSize={15}
+        >
+          <ArrowBackIcon /> back{' '}
+        </Text>
+        <DrawerCloseButton pos="static" onClick={handleCloseBasket} />
+      </DrawerHeader>
+
+      <DrawerBody>
+        <Text fontSize={18} fontWeight={600} mb={5}>
+          Payment Information
+        </Text>
+        <Flex flexDir="column" gap={5}>
+          <Text fontSize={15} fontWeight={600}>
+            Your voucher
+          </Text>
+          <Flex gap={4} align="start" mb={4} justify="space-between">
+            <Input
+              value={voucher.voucherKey || voucherCodeFromStorage || ''}
+              onInput={(e) => {
+                const target = e.target as HTMLInputElement
+                setVoucher({ voucherKey: target.value })
+              }}
+              placeholder="voucher"
+              w="40%"
+            />
+            <Button
+              alignSelf="end"
+              w="30%"
+              border="2px solid"
+              borderColor="turquoise.77"
+              bg="none"
+              borderRadius={25}
+              isDisabled={(isValid || isVoucherValid) ?? undefined}
```

Comment:
please rework handling flow: instead undefined should be null, probably typify state incoming value.